### PR TITLE
chore: add widget guest regression coverage

### LINKED ISSUE #981: XA-0: Guest widget regression test net
## Dependencies
- Blocks: XA-2 (#983), XA-3 (hard precondition — merges first)
- Blocked by: none

## Summary
The chat widget bootstrap (`frontend/public/widget.js`) and the widget chat page (`frontend/src/components/widget/public-agent-chat-page.tsx`) currently have no targeted test coverage. Before upcoming changes to the widget bootstrap land, add a regression net so every future PR keeps the existing guest flow releasable.

## Scope
1. Vitest coverage for the guest chat page flow:
   - embed-ticket flow: `/api/widget/embed-ticket` (top-level) → iframe `/api/widget/auth` with ticket → guest token
   - direct-visit flow: `widget_key` only, no origin allowlist applied
   - lazy task creation on first message; task resume via localStorage key; start-screen state
   - fail-closed error paths (invalid key, domain not allowed, disabled widget)
2. Smoke tests for `widget.js`: minimal testability refactor (extract logic into an importable module consumed by the IIFE, or jsdom-load the script), covering: missing `data-widget-key` error, embed-ticket fetch success → iframe URL composition, fail-closed on non-OK ticket response.
3. Wire both into the frontend CI job.

## Acceptance criteria
- All flows above covered and green in CI.
- `widget.js` behavior byte-for-byte unchanged apart from the testability refactor.
- Coverage report shows the two shared files are exercised (they are the blast radius of future bootstrap changes).

### PR DESCRIPTION
**Pull request by AlexLiu190625**
## Summary

- Add widget-specific Vitest configuration and CI coverage for the bootstrap script and public agent chat page.
- Cover embed-ticket bootstrap, direct widget-key access, lazy first-message task creation, task resume, and start-screen behavior.
- Pin fail-closed behavior for missing keys, rejected tickets, disallowed domains, and disabled widgets.

## Why

The shared guest widget paths had no targeted regression net. This change protects the existing integration contract before further widget bootstrap work, without modifying production runtime behavior.

## Validation

- `npm run test:widget` — 15 tests passed
- `npm run test:widget:coverage` — 15 tests passed
- `npm run type-check`
- Targeted ESLint checks
- Randomized test-order runs across three seeds

## Notes

- Production source files are unchanged.
- The coverage adapter follows the repository existing Vitest 2.x line and therefore surfaces the same existing Vitest advisory; a major Vitest upgrade is outside this regression-only change.

Closes #981

### CODE REVIEW

File: frontend/src/components/widget/public-agent-chat-page.test.tsx [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+      taskId: null,
+      currentTask: null,
+      isHistoryLoading: false,
+      isProcessing: false,
+    }
+    app.rerender = null
+    app.provider = null
+    sessionStorage.clear()
+    fetchMock.mockReset()
+    vi.stubGlobal("fetch", fetchMock)
```
[the remark this answers is not in the document]
Comment (AlexLiu190625) [reply frontend/src/components/widget/public-agent-chat-page.test.tsx:185]: The widget tests already inherit `frontend/src/vitest.setup.ts`, which registers a global `beforeEach` that calls `localStorage.clear()`. `frontend/vitest.config.ts` loads that setup file, and the widget configuration preserves it through `mergeConfig`. Adding another clear in this file would duplicate the existing test-isolation boundary. The focused widget suite also passes all 14 tests.

File: frontend/src/components/widget/widget-bootstrap.test.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+describe("widget bootstrap", () => {
+  const fetchMock = vi.fn()
+  let currentScriptDescriptor: PropertyDescriptor | undefined
+
+  beforeEach(() => {
+    currentScriptDescriptor = Object.getOwnPropertyDescriptor(document, "currentScript")
+    document.head.innerHTML = ""
+    document.body.innerHTML = ""
+    localStorage.setItem("xagent_guest_id", "guest-fixed")
```
[the remark this answers is not in the document]
Comment (AlexLiu190625) [reply frontend/src/components/widget/widget-bootstrap.test.ts:36]: `frontend/src/vitest.setup.ts` already clears `localStorage` in a global `beforeEach` for every widget test. This local `beforeEach` then deliberately seeds the deterministic `xagent_guest_id`. An additional clear here would duplicate the shared setup rather than improve isolation.

File: frontend/vitest.widget.config.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        "src/components/widget/public-agent-chat-page.tsx",
+      ],
+      reporter: ["text", "json-summary"],
+      reportsDirectory: "coverage/widget",
+      thresholds: {
+        perFile: true,
+        statements: 1,
+        branches: 1,
+        functions: 1,
+        lines: 1,
```
[the remark this answers is not in the document]
Comment (AlexLiu190625) [reply frontend/vitest.widget.config.ts]: The 1% values are intentional per-file non-zero exercise gates for #981. The acceptance criterion is that both shared files are exercised, not that every branch and function reaches 100%. CI runs `npm run test:widget:coverage`; current coverage is 94.27% statements / 95.45% branches for `widget.js` and 83.71% statements / 61.11% branches for the chat page. Setting all thresholds to 100 would fail valid regression coverage and change the issue scope.

**Review by rogercloud — COMMENTED** [review]
## Code Review: chore: add widget guest regression coverage

### PR Summary
This PR adds Vitest regression coverage for two previously-untested guest widget surfaces: the guest bootstrap script (`frontend/public/widget.js`) and the public agent chat page (`frontend/src/components/widget/public-agent-chat-page.tsx`). It introduces two test files, a dedicated `frontend/vitest.widget.config.ts`, and CI wiring in `.github/workflows/ci.yml`. There are zero production code changes — `widget.js` is untouched byte-for-byte. This closes issue #981 ("XA-0: Guest widget regression test net").

### Approach Verdict
**Sound / acceptable-with-reservations.** Pure test-addition is the correct and simplest approach here. Loading `widget.js` into jsdom via `window.eval` is exactly the alternative to a "testability refactor" that issue #981 explicitly permitted, and it preserves the production script unchanged (satisfying that acceptance criterion). All of issue #981's acceptance criteria are met — embed-ticket flow, direct-visit flow, lazy task creation, task resume, start-screen, and the 4 fail-closed paths are covered — and CI is wired correctly (added to the `frontend-build` job before the build step, gated on non-draft PRs). v8 coverage correctly attributes the eval-loaded code. The only design-level reservation is that the coverage thresholds are set so low they do not function as a real regression gate (see New Finding 1).

### Re-Review Update Summary
This review covers both a fresh independent pass and verification of the 3 prior Gemini Code Assist bot comments. No code changes have been made since those comments — this remains a single-commit PR, and the author (AlexLiu190625) responded only in the review threads without pushing further commits. The prior-finding statuses below reflect independent verification of the author's rebuttals.

### Prior Findings Status (Gemini Code Assist bot)

1. **WAIVED** — Missing `localStorage.clear()` in `public-agent-chat-page.test.tsx`'s local `beforeEach`. The global `beforeEach` in `frontend/src/vitest.setup.ts:57-59` already clears localStorage before every test, and `vitest.widget.config.ts` inherits `setupFiles` unchanged via `mergeConfig`. Independently verified: the root-scope setup hook runs before the per-file hook every time. Author's rebuttal is factually accurate. Non-issue.

2. **WAIVED** — Same missing-`localStorage.clear()` pattern in `widget-bootstrap.test.ts:36`, where the local hook seeds `xagent_guest_id` without first clearing. Independently verified: the same global setup hook applies, root-scope-before-per-file ordering is guaranteed, and `widget.js` reads `guest_id` fresh from localStorage on each invocation with no module-level state that could leak. Non-issue.

3. **PARTIAL** — Coverage thresholds set to `1` (1%), with Gemini suggesting `100`. The author's characterization of issue #981's scope ("files are exercised," not "100% coverage") is accurate, and Gemini's suggested `100` would break CI today (branches/functions are only 50-62%). However, `1%` is also too weak to act as a real regression gate. This is the same underlying concern as New Finding 1 below — treated there as a single merged finding.

### New Findings (all minor)

**1. Coverage thresholds are effectively toothless** — `frontend/vitest.widget.config.ts:14-20`
The thresholds (`perFile: true`; statements/branches/functions/lines all = `1`) sit far below measured coverage (widget.js: 94.27% stmts / 95.45% branch / 50% funcs; chat page: 83.71% stmts / 61.11% branch / 62.5% funcs). A future PR could delete most of these tests and CI would still pass, undercutting the PR's stated goal of protecting the existing integration contract. As noted under the PARTIAL prior finding, `100` is not achievable today. Recommend a realistic floor that actually gates regressions while leaving a small margin under current numbers, e.g. `statements: 80, lines: 80, branches: 55, functions: 45`.

**2. Guest-id auto-generation branch is never exercised** — `frontend/public/widget.js:109-111`
The shared `beforeEach` in `frontend/src/components/widget/widget-bootstrap.test.ts:36` always pre-seeds `localStorage.setItem("xagent_guest_id", "guest-fixed")` before every test, so the `if (!guestId) { guestId = ...; localStorage.setItem(...) }` generation path is never hit. Suggest one test that starts with empty localStorage and asserts a guest id is generated and persisted.

**3. Two confusingly similar test names** — `frontend/src/components/widget/widget-bootstrap.test.ts:92` and `:104`
`"does not navigate the iframe when embed-ticket authorization is rejected"` (HTTP 403 response) vs `"does not navigate the iframe when embed-ticket authorization rejects"` (network-level fetch rejection). Both assert the same downstream effect via genuinely different trigger paths, so a failing-test list wouldn't reveal which broke. Suggest explicit names, e.g. `"...on a non-OK ticket response"` vs `"...on a network failure"`.

### Optional Note (out of scope — no action required)
The FAB open/close toggle (`frontend/public/widget.js:179-186`) has no test coverage. This is explicitly outside issue #981's smoke-test list (which requires only the missing-key error, embed-ticket → iframe URL composition, and fail-closed on non-OK response), so it is not a finding — just a genuine coverage gap worth noting for future widget bootstrap work.

### Testing
`npm run test:widget:coverage` was run and passed (14/14). Shuffled test-order runs (seeds 12345 and 999) were also verified stable, and v8 coverage was confirmed to correctly attribute the eval-loaded `widget.js`.

File: frontend/vitest.widget.config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (AlexLiu190625) [reply frontend/vitest.widget.config.ts]: Updated the per-file thresholds to statements 80, branches 55, functions 45, and lines 80. The full 15-test widget suite passes, while a reduced two-happy-path diagnostic now fails the per-file coverage gate. This preserves a meaningful regression floor without requiring unattainable 100% branch and function coverage.

**Review by rogercloud — APPROVED** [review]
## Follow-up review — PR #999 (`chore: add widget guest regression coverage`)

### Update summary
New commit `ae7ad6a4d` ("chore: strengthen widget regression coverage") addresses all 3 findings from the prior review. This follow-up re-verifies each fix against the code and re-runs a full independent pass over the diff (base `e9b6920c` → head `ae7ad6a4`).

### Prior findings — status

| # | Prior finding | Status | Evidence |
|---|---------------|--------|----------|
| 1 | Coverage thresholds set to a no-op `1/1/1/1` (also the Gemini bot's PARTIAL note) | **FIXED** | `vitest.widget.config.ts` now `statements:80 / branches:55 / functions:45 / lines:80`, `perFile: true`; measured coverage clears all four on both files (tightest margin chat-page statements/lines +3.71pp). |
| 2 | Missing coverage for guest-id auto-generation | **FIXED** | New test clears `localStorage`, mocks `Math.random` to match the two-call generation logic, and exercises the previously-uncovered branch at `widget.js:109-111`; widget.js branch coverage is now 100%. |
| 3 | Two confusingly similar test names | **FIXED** | Pure rename ("...on a non-OK embed-ticket response" / "...on an embed-ticket network failure"); bodies unchanged, both pass. |

### Round 0 verdict
Sound. Pure test-addition — `widget.js` is byte-for-byte unchanged from base, all of #981's acceptance criteria are met, and the coverage thresholds are now a meaningful CI gate. No remaining design-level concerns.

### New findings (both low severity, non-blocking)

**A. Dead mock in the new guest-id test — `widget-bootstrap.test.ts:64-65`**
The `vi.spyOn(Math, "random").mockReturnValueOnce(0.123456).mockReturnValueOnce(0.654321)` lines have no effect on the assertion: the test only matches the generic regex `/^guest_[a-z0-9]+$/` and interpolates whatever value is produced, so removing the mock leaves the test passing identically. Either assert the exact deterministic value the mock produces (`guest_4fzyo82mvyqnk000qvgin`) to actually lock down the generation logic — consistent with the PR's byte-for-byte spirit — or drop the now-pointless mock.

**B. `test:widget` doesn't reproduce the CI gate — `package.json:13`**
`test:widget` runs under the default `vitest.config.ts` (no `--config vitest.widget.config.ts`), so it never applies the coverage thresholds that `test:widget:coverage` (used in CI) enforces. Both scripts run the same two test files and both pass — this is a conventional `test` / `test:coverage` split, not a bug — but a developer running `npm run test:widget` locally to check "does this pass the gate" would get a false sense of completeness. FYI only.

Neither item blocks merge.

### Testing
15/15 tests pass. Coverage thresholds were verified empirically to be a real gate: trimming the suite to two happy-path tests makes `test:widget:coverage` fail with explicit threshold errors on both files. Test isolation, mock/fixture correctness (auth-payload key omission for embed vs. direct-visit flows, storage keys, `onSend` parameter mapping), CI wiring, and dependency changes were all re-verified with no new problems.
